### ISSUE TITLE
Stop logging my data directory path in home page in case of error

### ISSUE THREAD

**Issue by LeonardoGentile** [nextcloud/server#2046]
Hello,
honestly I have no idea what to do with issues that I've opened for owncloud but that also affect nextcloud directly.
I will replicate https://github.com/owncloud/core/issues/26386 here, fell free to close it if you have already a plan to port issues from there automatically.

### Steps to reproduce
1. store the user data on an external hard disk
2. remove the hard disk or unmount it
3. open the oc home page
### Expected behaviour

It should tell me there is a configuration error
### Actual behaviour

The home page (available to everyone, logged or anonymous users) tells me there is a configuration error (the Data Directory is invalid) BUT it also tells everyone the path of the data directory. This is both a privacy and security concerns. I'm not sure if intentional or not, but I would disable the print of the full path in home page. I could not find any configuration how to disable this behaviour

```
Data directory (/full/path/to/data) is invalid
Please check that the data directory contains a file ".ocdata" in its root.
```
### Server configuration

**Operating system**: 
debian

**Web server:** 
apache

**Database:** 
mysql

**PHP version:** 
5

**ownCloud version:** (see ownCloud admin page)
9.1

**Updated from an older ownCloud or fresh install:**
fresh install
**Where did you install ownCloud from:**
From the repository
**Signing status (ownCloud 9.0 and above):**

**The content of config/config.php:**

```
$CONFIG = array (
   'instanceid' => 'my.id',
   'passwordsalt' => 'pepper',
   'secret' => 'my.secret',
   'trusted_domains' =>
   array (
       0 => 'my.oc.com',
   ),
   'datadirectory' => '/full/path/to/data',
   'overwrite.cli.url' => 'https://my.oc.com',
   'dbtype' => 'mysql',
   'version' => '9.1.0.15',
   'dbname' => 'owncloud',
   'dbhost' => 'localhost',
   'dbtableprefix' => 'own_',
   'dbuser' => 'mydbuser',
   'dbpassword' => '***REMOVED SENSITIVE VALUE***',
   'logtimezone' => 'Europe/Paris',
   'logfile' => '/var/log/owncloud.log',
   'loglevel' => '2', // Warning
   'installed' => true,
   'cron_log' => true,
   'memcache.local' => '\OC\Memcache\APCu',
 );
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
Local on an external hard disk
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

**Comment by nickvergessen** [comment]
Since this is nothing the user can "cause", I'm not sure if it is worth "fixing" this.
I guess there are more people seeing the message which afterwards need the path to fix something on the setup, so hiding the path is creating "troubles".
But I also saw that this message is displayed even in maintenance mode.

However hiding doesn't work very well, since we can't even log the error because default log path is in the data dir...

**Comment by LeonardoGentile** [comment]
A natural way to handle this would be to print the error to the log file. Maybe you could print this in the web page if in "debug/dev" mode but I'm not sure if there is a similar concept in the nextcloud. Anyway of course if the data dir is missing it would be impossible to print it there. So while we are on this, I think logging to the data dir by default is a terrible idea (take this example) it would feel more natural to move the default log file into the installation dir or in a subdir "log" as similar to what seafile does.

**Comment by nickvergessen** [comment]
But then the root would be required to be writable...

We have a config for debug, yeah could do that.

**Comment by harry-7** [comment]
I would like to work on this issue

**Comment by MorrisJobke** [comment]
Fix is in #4210
